docs/how-to-guides/customize-templates.md

```diff
@@ -0,0 +1,134 @@
+This how-to-guide shows you how to customize the liquid templates for the FHIR Converter service.
+
+The default templates for the FHIR Converter service are located [here](https://github.com/microsoft/FHIR-Converter/tree/main/data/), and are a recommended starting point for creating customized templates.
+
+## Transforming data to FHIR
+
+### HL7v2 to FHIR
+When creating a custom template for HL7v2 to FHIR, it may be simplest to start creating a template that generates a FHIR bundle with only the FHIR resource(s) needed for your use case. The default templates iterate through the HL7v2 segments and make a best effort to generate FHIR resources as applicable, but for starting a custom template for an HL7v2 ADT_A01 message, for example, you may want to focus on generating resources like Patient, Encounter, and Coverage, and then later add additional resources as needed.
+
+This [page](https://github.com/microsoft/FHIR-Converter/blob/main/docs/HL7v2-templates.md) contains information on what segments are mapped for each HL7v2 message type when using the default templates. You may want to decide on what is necessary to include in the output FHIR bundle, and then remove any content from the default template that is not necessary for your use case. 
+
+In addition, there are some pages dedicated to HL7v2 to FHIR reusable snippets, and HL7v2 to FHIR important points, which may be useful when creating custom templates.
+ - [Snippet concepts](https://github.com/microsoft/FHIR-Converter/blob/shared/convert-api-documentation-update/docs/SnippetConcept.md)
+ - [HL7v2 to FHIR important points](https://github.com/microsoft/FHIR-Converter/blob/shared/convert-api-documentation-update/docs/HL7v2-ImportantPoints.md)
+
+### CCD to FHIR
+The default templates are a good starting point for creating a custom CCD transformation template. A FHIR bundle is created by iterating over the common CCD sections in this [template](https://github.com/microsoft/FHIR-Converter/blob/main/data/Templates/Ccda/CCD.liquid). As a starting point, you can start with the CCD sections that are needed and then modify the subtemplates referenced by this root template.
+
+### JSON to FHIR
+There are some [sample templates](https://github.com/microsoft/FHIR-Converter/tree/main/data/Templates/Json) that map JSON values to a Patient resource.
+
+## Transforming from FHIR to HL7v2
+When transforming FHIR to HL7v2, the templates are written such that the input FHIR resource is transformed into a JSON representation of an HL7v2 message. When the /convertToHl7v2 API is called, the API will transform the input FHIR resource into the JSON representation of an HL7 message. But before the result is returned to the user, the JSON representation of an HL7v2 message is transformed into the normal HL7v2 format with line breaks, and is returned to the user.
+
+To represent HL7v2 in JSON, the transformation uses an array called messageDefinition, which is an ordered array of HL7v2 message segments.
+
```

Comment:
nit: can we also add some doc around Hl7v2 spec here for reference.

Reply:
added a link to hl7.org which has the HL7v2 specifications. Didn't want to include a deep link because the site changes from time to time